specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,764 @@
+
+ ```cpp
+    webview2->add_ContextMenuRequested(
+        Callback<ICoreWebView2ContextMenuRequestedEventHandler>(
+            [this](
+                ICoreWebView2* sender,
+                ICoreWebView2ContextMenuRequestedEventArgs* args)
+            {
+                auto showMenu = [this, args]
+                {
+                    wil::com_ptr<ICoreWebView2ContextMenuItemCollection> items;
+                    CHECK_FAILURE(args->get_MenuItems(&items));
+                    CHECK_FAILURE(args->put_Handled(true));
+                    HMENU hPopupMenu = CreatePopupMenu();
+                    AddMenuItems(hPopupMenu, items);
+                    HWND hWnd;
+                    m_controller->get_ParentWindow(&hWnd);
+                    SetForegroundWindow(hWnd);
+                    wil::com_ptr<ICoreWebView2ContextMenuInfo> parameters;
+                    CHECK_FAILURE(args->get_ContextMenuInfo(&parameters));
+                    POINT locationInControlCoordinates;
+                    POINT locationInScreenCoordinates;
+                    CHECK_FAILURE(parameters->get_Location(&locationInControlCoordinates));
+                    // get_Location returns coordinates in relation to upper left Bounds of the WebView2.Controller. Will need to convert to Screen coordinates to display the popup menu in the correct location.
+                    ConvertToScreenCoordinates(locationInControlCoordinates, locationInScreenCoordinates);
+                    UINT32 selectedCommandId = TrackPopupMenu(hPopupMenu, TPM_TOPALIGN | TPM_LEFTALIGN | TPM_RETURNCMD, locationInScreenCoordinates.x, locationInScreenCoordinates.y, 0, hWnd, NULL);
+                    CHECK_FAILURE(args->put_SelectedCommandId(selectedCommandId));
+                };
+                wil::com_ptr<ICoreWebView2Deferral> deferral;
+                CHECK_FAILURE(args->GetDeferral(&deferral));
+                m_sampleWindow->RunAsync([deferral, showMenu]() {
+                    showMenu();
+                    CHECK_FAILURE(deferral->Complete());
+                });
+                return S_OK;
+            }).Get(),
+        &m_contextMenuRequestedToken);
+
+    void ContextMenu::AddMenuItems(
+        HMENU hPopupMenu, wil::com_ptr<ICoreWebView2ContextMenuItemCollection> items)
+        {
+            wil::com_ptr<ICoreWebView2ContextMenuItem> current;
+            UINT32 itemsCount;
+
+            CHECK_FAILURE(items->get_Count(&itemsCount));
+            for (UINT32 i = 0; i < itemsCount; i++)
+            {
+                CHECK_FAILURE(items->GetValueAtIndex(i, &current));
+                COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND kind;
+                CHECK_FAILURE(current->get_Kind(&kind));
+                wil::unique_cotaskmem_string label;
+                CHECK_FAILURE(current->get_Label(&label));
+                std::wstring labelString = label.get();
+                BOOL isEnabled;
```

Comment:
We didn't do a `get_ShortcutKeyDescription` and append it to the label (with a `\t` in between).